packages/dapp/src/hooks/useKeysManager.ts

```diff
@@ -0,0 +1,212 @@
+import type { VerificationMethodReference } from '@windingtree/org.json-schema/types/org.json';
+import type { AnySchema } from '@windingtree/org.id-utils/dist/object';
+import type { GenericStateRecord } from '../store';
+import { useState, useCallback, useEffect } from 'react';
+import { useAppReducer } from '../store/reducer';
+
+import { object, uid } from '@windingtree/org.id-utils';
+import { org } from '@windingtree/org.json-schema';
+
+export const revocationReasonValues = object.getDeepValue(org, 'definitions.VerificationMethodRevocationReference.properties.reason.enum') as string[];
+export type RevocationReference = typeof revocationReasonValues[number]
+
+export const allowedKeysTypes: string[] = ['EcdsaSecp256k1VerificationKey2019'];
+
+export type KeyType =
+	typeof allowedKeysTypes[number];
+
+const keyRecordSchema: AnySchema = {
+  type: 'object',
+  properties: {
+    id: {
+      type: 'string'
+    },
+    type: {
+      type: 'string',
+      enum: allowedKeysTypes
+    },
+    publicKey: {
+      type: 'string'
+    },
+    tag: {
+      type: 'string'
+    },
+    note: {
+      type: 'string'
+    },
+    revocation: {
+      type: 'object',
+      properties:{
+        reason: {
+          type: 'string',
+          enum: revocationReasonValues
+        },
+        invalidityDate: {
+          type: 'string',
+          format: 'data-time'
+        },
+      },
+      required: ['reason', 'invalidityDate']
+    }
+  },
+  required: ['id', 'type', 'publicKey', 'tag']
+}
+
+export interface KeyRecord extends GenericStateRecord {
+  type: KeyType;
+	publicKey: string; // Ethereum account address
+	tag: string; // Unique across records formatted string
+	note?: string; // Key description
+	revocation?: VerificationMethodReference['verificationMethodRevocation']; // see this type for details
+}
+
+export type Keys = KeyRecord[];
+
+export type UseKeysManagerHook = [
+  addKey: Function,
+  updateKey: Function,
```

Comment:
When used declaration such as just `Function` this prevents input arguments validation from the Typescript side.
Sometimes it is acceptable. But it is better to define a propper type definition.
In this case, it can be:
```typescript
updateKey: (record: KeyRecord) => void,
```
for next two lines the same